### PR TITLE
[FIDGET] Channel Fidget

### PR DESCRIPTION
**Pull request by gitpancake**
This is a channel fidget!

Two hooks - get channel info / get channel casts
Limit: 20 casts per feed 
Enter channel ID, the feed fetches all information and displays image / text content in a feed

WIP:

Lazy loading on scroll
Clean up CSS

Wallet address (base): 0x412B59e147aF4513c2EA537cE5036c357f2A44a7
Wallet address (mainnet): 0xaa3c0FAa24E67eBe9B73Fe3efaf9D06F9Ba57622

Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,8 @@ pids
 *.pid.lock
 .DS_Store
 
+.env**
```
Comment (hiporox) [inline .gitignore:19]: Please remove this. `.env` files should be included, following the Next.js spec for local vs. share variables: https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables

We already include the following lower on in the the `.gitignore`:
```
# dotenv environment variables file
.env.*.local
.env.test
.env.local
```

File: README.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 Forked from [herocast](https://github.com/hellno/herocast/)
 
+# Docs
+https://docs.nounspace.com/nounspace-alpha/
```
Comment (hiporox) [inline README.md:7]: Great, thank you!

File: src/common/ui/templates/space/index.tsx [hunk 3]
```diff
@@ -1,31 +1,26 @@
-import DefaultGrid from "@/fidgets/layout/grids/defaultGrid";
-import Gallery from "@/fidgets/ui/gallery";
-import Feed from "@/pages/feed";
-import GridLayout from "react-grid-layout";
-import React, { ReactElement, useState } from 'react';
 import _ from "lodash";
+import { ReactElement } from "react";
```
Comment (hiporox) [inline src/common/ui/templates/space/index.tsx:2]: `React` needs to be imported into all `.tsx` files

File: src/common/ui/templates/space/index.tsx [hunk 4]
```diff
@@ -41,11 +36,11 @@ export default function Space({ config, isEditable, children }: SpaceArgs){
   return (
     <div className="m-4">
       <ReactGridLayout onLayoutChange={onLayoutChange} {...config.layoutConfig}>
-        <div key={0} data-grid={config.fidgetConfigs[0]} className="overflow-hidden rounded-md flex justify-center items-center">
+        {/* <div key={0} data-grid={config.fidgetConfigs[0]} className="overflow-hidden rounded-md flex justify-center items-center">
```
Comment (hiporox) [inline src/common/ui/templates/space/index.tsx:39]: Why did this get commented out?

**Comment by hiporox** [comment]
@gitpancake This is awesome. Thank you for your contribution! 

I have updated the base branch so that I can better compare the changes you made to @j-paterson `LayoutSystem` changes. I will add some reviews to this PR shortly

File: src/common/ui/templates/space/index.tsx [hunk 5]
```diff
@@ -1,31 +1,26 @@
...
+import { ReactElement } from "react";
 import RGL, { WidthProvider } from "react-grid-layout";
 
 type SpaceArgs = {
   config: {
     fidgetConfigs: {
-      [key: string]: any
-    }
+      [key: string]: any;
+    };
```
Comment (hiporox) [inline src/common/ui/templates/space/index.tsx:9]: Thank you for making is correct Typescript declarations

**Review by hiporox — CHANGES_REQUESTED** [review]
Two major things beyond the little comments:

1. You have placed all of your code into the `src/fidgets/example/index.tsx` and created a new file `src/fidgets/channel/index.tsx` that has the old content of `src/fidgets/example/index.tsx`. Please change this so that all of your code is in the new file `src/fidgets/channel/index.tsx` and leave the example file alone

2. You have a bunch of lint changes that are applied to `src/pages/homebase/index.tsx`. Please remove these from the PR. I am happy to discuss the lint style settings, but that should be reserved for a different PR

**Comment by hiporox** [comment]
@gitpancake I have rebased your changes into the [feat/channel-fidget](https://github.com/Nounspace/nounspace.ts/tree/feat/channel-fidget) branch. Feel free to cherry pick/copy from there to resolve merge conflicts that now exist

**Comment by gitpancake** [comment]
Hey @hiporox  thank you for the review! I will review and make changes this week

**Comment by hiporox** [comment]
Sounds good. I know we did a pretty massive overhaul of the system since you last looked at it. Let me know if you need help porting to the new system

**Pull request closed without merging**